abcclassroom/utils.py

```diff
@@ -132,28 +133,6 @@ def get_editor():
     return os.environ.get("VISUAL") or os.environ.get("EDITOR") or "vi"
 
 
-def _call_git(*args, directory=None):
-    cmd = ["git"]
-    cmd.extend(args)
-    try:
-        ret = subprocess.run(
-            cmd,
-            cwd=directory,
-            check=True,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-    except subprocess.CalledProcessError as e:
-        err = e.stderr.decode("utf-8")
-        if err:
-            msg = err.split(":")[1].strip()
-        else:
-            msg = e.stdout.decode("utf-8")
-        raise RuntimeError(msg) from e
-
-    return ret
-
-
 @lru_cache(1)
```

Comment:
@kcranston can you help me understand this decorator? i'm newer to these !!

Reply:
(this is from Tim) It is simply caching the result because this TOP function got called a lot in old abc-classroom. See https://docs.python.org/3/library/functools.html